pkg/network/ebpf/c/protocols/http/buffer.h

```diff
@@ -33,6 +33,22 @@ static __always_inline void read_into_buffer(char *buffer, char *data, size_t da
     }
 }
 
+static __always_inline void read_into_buffer_classification(char *buffer, char *data, size_t data_size) {
+    bpf_memset(buffer, 0, CLASSIFICATION_MAX_BUFFER);
+
+    // we read CLASSIFICATION_MAX_BUFFER-1 bytes to ensure that the string is always null terminated
+    if (bpf_probe_read_user_with_telemetry(buffer, CLASSIFICATION_MAX_BUFFER - 1, data) < 0) {
+// note: arm64 bpf_probe_read_user() could page fault if the CLASSIFICATION_MAX_BUFFER overlap a page
+#pragma unroll(CLASSIFICATION_MAX_BUFFER - 1)
+        for (int i = 0; i < CLASSIFICATION_MAX_BUFFER - 1; i++) {
+            bpf_probe_read_user(&buffer[i], 1, &data[i]);
+            if (buffer[i] == 0) {
+                return;
+            }
+        }
+    }
+}
```

Comment:
isn't it better to calculate the page overlap scenario (you can use the buffer address to estimate related position in the page) and perform the read using `Min(CLASSIFICATION_MAX_BUFFER, Page_Size-(address_of_the_buffer % Page_Size))`

Reply:
I didn't want to mix between the different PRs